src/plugins/irc-events/link.js

```diff
@@ -98,12 +98,14 @@ function parse(msg, preview, res, client) {
 	case "image/jpeg":
 	case "image/webp":
 		if (res.size > (Helper.config.prefetchMaxImageSize * 1024)) {
-			return;
+			preview.type = "error";
+			preview.head = "Large image";
```

Comment:
I don't think errors need head, just make it body in red color.

Reply:
It felt a bit odd without a head, to be honest. I'm not sure why. Let's see... Red text or red background? Thoughts?

|Red text|Red background|
|-|-|
|![](https://files.yamanickill.com/jjYHBaCI.png)|![](https://files.yamanickill.com/72oTqtrQ.png)|

Reply:
Red background draws too much attention for no good reason.

Reply:
Also not sure it should be “your max image size”, as the client may not own the server / instance.

Reply:
Yeah, I knew I wouldn't have the correct wording. Not sure of the best way to word this.

We want the user to know that this is an option on the server that can be changed, but if they aren't the admin, we don't want them to be confused.

"Image is larger than the max image size. Click to open"

That doesn't give them any indication as to how to change that (or ask their admin to change it)

Then again

"Image is larger than the server's max prefetch size. Click to open" 

That's clearly too technical. (Also isn't clear which server it's talking about, necessarily).

Reply:
I did that some time ago and @xPaw didn't want to display anything on images that were too big. Wish I didn't kill the commit 😢

Reply:
> We want the user to know that this is an option on the server that can be changed

I don't agree with this actually. Docs + clear configuration + potentially server logs (probably not in this case) should help server admins as much as possible, but I think any assumption that user might also be the admin will lead to confusing statements. I think a longer-term solution is to have users with an admin role (but this is a different story :D).

Anyway, in the meantime, I propose this:

<img width="799" alt="screen shot 2017-12-28 at 16 30 18" src="https://user-images.githubusercontent.com/113730/34423844-1317e0b4-ebed-11e7-9476-182cf019c391.png">

@YaManicKill, I'm adding a commit with changes I made to get there, but this is purely for convenience so you don't have to copy-paste things around. Feel free to trash it if you don't like it (I do, surprising huh? 😁) and squash it if you do.

If/when we have admin role (think creating users from the interface, listing users on a TL instance, messaging all users on this instance, stats, etc.), we can add an extra line that says: "Admin: Current max size is XXX MB, you may change this value in your `config.js` file."

Reply:
Yeah sure, looks better.

> If/when we have admin role

I thought you didn't want this :-P I am very much loving the idea of doing this though, for the record.

Reply:
> I thought you didn't want this :-P I am very much loving the idea of doing this though, for the record.

I don't remember being against this, no :)
Every time we assume user == admin, I see a use case for it! Also, it would be a convenient way to say "this user also is an admin on the server" (even though it doesn't have to technically be true, we can safely assume this).
I'll draft something for v3, it would be a good paradigm shift to introduce.

Anyway, didn't mean to hijack this PR with an unrelated feature, sorry.